src/main/java/io/apicurio/datamodels/core/validation/rules/other/SecurityRequirementUniquenessValidationRule.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2019 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.datamodels.core.validation.rules.other;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import io.apicurio.datamodels.core.Constants;
+import io.apicurio.datamodels.core.models.common.ISecurityRequirementParent;
+import io.apicurio.datamodels.core.models.common.SecurityRequirement;
+import io.apicurio.datamodels.core.validation.ValidationRule;
+import io.apicurio.datamodels.core.validation.ValidationRuleMetaData;
+
+/**
+ * @author cfoskin@redhat.com
+ */
+public class SecurityRequirementUniquenessValidationRule extends ValidationRule {
+
+    /**
+     * Constructor.
+     * 
+     * @param ruleInfo
+     */
+    public SecurityRequirementUniquenessValidationRule(ValidationRuleMetaData ruleInfo) {
+        super(ruleInfo);
+    }
+
+    /**
+     * @param currentSecurityRequirement
+     * @param otherSecurityRequirement
+     * @return returns true if the two security requirements are equal
+     */
+    private boolean isEqualTo(SecurityRequirement currentSecurityRequirement,
+            SecurityRequirement otherSecurityRequirement) {
```

Comment:
Let's make this method `static` since it doesn't access any internal state and only operates on its arguments.